core/src/main/java/hudson/WebAppMain.java

```diff
@@ -367,6 +368,7 @@ public FileAndDescription getHomeDir(ServletContextEvent event) {
 
     public void contextDestroyed(ServletContextEvent event) {
         try {
+            ACL.impersonate(ACL.SYSTEM);
```

Comment:
Not actually doing anything for us here (since Jenkins dies after), but if it'll ease things, sure.